### ISSUE TITLE
interp1d(kind='quadratic') fails on very large ints

### ISSUE THREAD

**Issue by crusaderky** [scipy/scipy#8782]
scipy 1.0.1, Anaconda 5, Linux 64 bit

Trying to do an interpolation on dates:
```
N_YEARS = 120
x = pandas.to_datetime('2015-12-31') + pandas.to_timedelta(numpy.arange(0, N_YEARS*365, 365), 'D')
x = numpy.array(x).astype('M8[ns]').astype(int)
xnew = pandas.to_datetime('2016-01-10')
xnew = xnew.to_datetime64().astype('M8[ns]').astype(int)
y = numpy.linspace(0, 1, x.size)

print(x)
print(y)
print(xnew)

scipy.interpolate.interp1d(x, y, kind='quadratic')(xnew)
```
Output:
```
[1451520000000000000 1483056000000000000 1514592000000000000
 1546128000000000000 1577664000000000000 1609200000000000000
 1640736000000000000 1672272000000000000 1703808000000000000
 1735344000000000000 1766880000000000000 1798416000000000000
 1829952000000000000 1861488000000000000 1893024000000000000
 1924560000000000000 1956096000000000000 1987632000000000000
 2019168000000000000 2050704000000000000 2082240000000000000
 2113776000000000000 2145312000000000000 2176848000000000000
 2208384000000000000 2239920000000000000 2271456000000000000
 2302992000000000000 2334528000000000000 2366064000000000000
 2397600000000000000 2429136000000000000 2460672000000000000
 2492208000000000000 2523744000000000000 2555280000000000000
 2586816000000000000 2618352000000000000 2649888000000000000
 2681424000000000000 2712960000000000000 2744496000000000000
 2776032000000000000 2807568000000000000 2839104000000000000
 2870640000000000000 2902176000000000000 2933712000000000000
 2965248000000000000 2996784000000000000 3028320000000000000
 3059856000000000000 3091392000000000000 3122928000000000000
 3154464000000000000 3186000000000000000 3217536000000000000
 3249072000000000000 3280608000000000000 3312144000000000000
 3343680000000000000 3375216000000000000 3406752000000000000
 3438288000000000000 3469824000000000000 3501360000000000000
 3532896000000000000 3564432000000000000 3595968000000000000
 3627504000000000000 3659040000000000000 3690576000000000000
 3722112000000000000 3753648000000000000 3785184000000000000
 3816720000000000000 3848256000000000000 3879792000000000000
 3911328000000000000 3942864000000000000 3974400000000000000
 4005936000000000000 4037472000000000000 4069008000000000000
 4100544000000000000 4132080000000000000 4163616000000000000
 4195152000000000000 4226688000000000000 4258224000000000000
 4289760000000000000 4321296000000000000 4352832000000000000
 4384368000000000000 4415904000000000000 4447440000000000000
 4478976000000000000 4510512000000000000 4542048000000000000
 4573584000000000000 4605120000000000000 4636656000000000000
 4668192000000000000 4699728000000000000 4731264000000000000
 4762800000000000000 4794336000000000000 4825872000000000000
 4857408000000000000 4888944000000000000 4920480000000000000
 4952016000000000000 4983552000000000000 5015088000000000000
 5046624000000000000 5078160000000000000 5109696000000000000
 5141232000000000000 5172768000000000000 5204304000000000000]
[0.         0.00840336 0.01680672 0.02521008 0.03361345 0.04201681
 0.05042017 0.05882353 0.06722689 0.07563025 0.08403361 0.09243697
 0.10084034 0.1092437  0.11764706 0.12605042 0.13445378 0.14285714
 0.1512605  0.15966387 0.16806723 0.17647059 0.18487395 0.19327731
 0.20168067 0.21008403 0.21848739 0.22689076 0.23529412 0.24369748
 0.25210084 0.2605042  0.26890756 0.27731092 0.28571429 0.29411765
 0.30252101 0.31092437 0.31932773 0.32773109 0.33613445 0.34453782
 0.35294118 0.36134454 0.3697479  0.37815126 0.38655462 0.39495798
 0.40336134 0.41176471 0.42016807 0.42857143 0.43697479 0.44537815
 0.45378151 0.46218487 0.47058824 0.4789916  0.48739496 0.49579832
 0.50420168 0.51260504 0.5210084  0.52941176 0.53781513 0.54621849
 0.55462185 0.56302521 0.57142857 0.57983193 0.58823529 0.59663866
 0.60504202 0.61344538 0.62184874 0.6302521  0.63865546 0.64705882
 0.65546218 0.66386555 0.67226891 0.68067227 0.68907563 0.69747899
 0.70588235 0.71428571 0.72268908 0.73109244 0.7394958  0.74789916
 0.75630252 0.76470588 0.77310924 0.78151261 0.78991597 0.79831933
 0.80672269 0.81512605 0.82352941 0.83193277 0.84033613 0.8487395
 0.85714286 0.86554622 0.87394958 0.88235294 0.8907563  0.89915966
 0.90756303 0.91596639 0.92436975 0.93277311 0.94117647 0.94957983
 0.95798319 0.96638655 0.97478992 0.98319328 0.99159664 1.        ]
1452384000000000000

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-89-076d28a4835e> in <module>()
      1 import scipy.interpolate
----> 2 scipy.interpolate.interp1d(x, y, kind='quadratic')(xnew)

~/anaconda3/lib/python3.6/site-packages/scipy/interpolate/interpolate.py in __init__(***failed resolving arguments***)
    518 
    519             self._spline = make_interp_spline(xx, yy, k=order,
--> 520                                               check_finite=False)
    521             if rewrite_nan:
    522                 self._call = self.__class__._call_nan_spline

~/anaconda3/lib/python3.6/site-packages/scipy/interpolate/_bsplines.py in make_interp_spline(x, y, k, t, bc_type, axis, check_finite)
    753         raise ValueError("Expect non-negative k.")
    754     if t.ndim != 1 or np.any(t[1:] < t[:-1]):
--> 755         raise ValueError("Expect t to be a 1-D sorted array_like.")
    756     if x.size != y.shape[0]:
    757         raise ValueError('x and y are incompatible.')

ValueError: Expect t to be a 1-D sorted array_like.
```
Things that make the problem disappear:
- replace to ``.astype(M8[ns]).astype(int)`` with ``.astype(M8[D]).astype(int)``, thus obtaining much smaller ints
- replace to ``.astype(M8[ns]).astype(int)`` with ``.astype(M8[ns]).astype(float)``
- reduce N_YEARS below 100 (I have no idea about why)
- change ``kind='quadratic'`` to any other algorithm

**Comment by pv** [comment]
It won't interpolate at integer x-axis, it will first cast them to floats and then work.

**Comment by pv** [comment]
The above code actually runs fine for me also with integers.

**Comment by crusaderky** [comment]
Attaching my exact package config
[conda_list.txt](https://github.com/scipy/scipy/files/1958037/conda_list.txt)

**Comment by ev-br** [comment]
`kind="quadratic"` being special suggests it's a relative of https://github.com/scipy/scipy/issues/8714, which has been fixed recently (the fix will be available in scipy 1.1.0, but is not available in 1.0.1).

And indeed, the OP example works for me with 

```
In [5]: scipy.__version__
Out[5]: '1.1.0.dev0+3fd28b5'
```

Can you double-check that the problem goes away with scipy 1.1.0rc1?

**Comment by crusaderky** [comment]
Confirmed in 1.1.0 the problem disappears, thanks
